dns-global-exclusive-tls.ks.in

```diff
@@ -0,0 +1,102 @@
+# Test the e2e functionality of global exclusive encrypted DNS functionality
+# Partial tests of reqiured features are in 'certificate' and 'dns-global-bootopts'
+# tests.
+
+# Use defaults, but no network
+%ksappend common/common.ks
+%ksappend repos/default.ks
+%ksappend l10n/default.ks
+%ksappend users/default.ks
+%ksappend storage/default.ks
+%ksappend payload/default_packages.ks
+
+%certificate --filename tls-ca-bundle.pem --dir /etc/pki/dns/extracted/pem
+%ksappend data/edns-idmops.psi.redhat.com.crt
+%end
+
+# Reboot the installed system.
+reboot
+
+# Validate on the first boot.
+%ksappend validation/success_on_first_boot.ks
+
+
+%pre
+
+# Check the presence in installer environment in %pre
+if [ ! -f /etc/pki/dns/extracted/pem/tls-ca-bundle.pem ]; then
+    echo "*** tls-ca-bundle.pem not found in /etc/pki/dns/extracted/pem in installer environment in %pre" >> /root/RESULT
+fi
+
+HOSTNAME=fedoraproject.org
+nslookup ${HOSTNAME}
+if [[ $? -ne 0 ]]; then
+    echo "*** Failed check: name resolution on ${HOSTNAME} failed in %pre" >> /root/RESULT
+fi
+
+%end
+
+%post --nochroot
+
+@KSINCLUDE@ post-nochroot-lib-network.sh
+# First copy result from %pre stage to chroot
+copy_pre_stage_result
+
+# Check the presence in installer environment during %post
+if [ ! -f /etc/pki/dns/extracted/pem/tls-ca-bundle.pem ]; then
+    echo "*** tls-ca-bundle.pem not found in /etc/pki/dns/extracted/pem in installer environment %post --nochroot" >> /mnt/sysroot/root/RESULT
+fi
+
+HOSTNAME=fedoraproject.org
+nslookup ${HOSTNAME}
+if [[ $? -ne 0 ]]; then
+    echo "*** Failed check: name resolution on ${HOSTNAME} failed in %post --nochroot" >> /mnt/sysroot/root/RESULT
+fi
+
+%end
+
+%post
+
+# Check the presence on installed system
+if [ ! -f /etc/pki/dns/extracted/pem/tls-ca-bundle.pem ]; then
+    echo "*** tls-ca-bundle.pem not found in /etc/pki/dns/extracted/pem on installed system" >> /root/RESULT
+fi
+
+%ksappend validation/success_if_result_empty.ks
```

Comment:
Shouldn't it be the last command in this `%post` section?
Or, is this validation needed at all in this phase, since validation is done on the installed system by the `success_on_first_boot.ks` fragment?

Reply:
Oh it should, nice catch.

Reply:
Updated